pkg/api/controller.go

```diff
@@ -697,6 +704,25 @@ func (c *Controller) CreatePolicy(w http.ResponseWriter, r *http.Request, body C
 	ctx := r.Context()
 	c.LogAction(ctx, "create_policy", r, "", "", "")
 
+	// Check that policy ID is valid
+	valid, msg := c.isNameValid(body.Id, "Policy")
+	if !valid {
+		writeError(w, http.StatusBadRequest, msg)
+		return
+	}
+	
+	// Check that we don't already have a policy with this ID
+	_, gpErr := c.Auth.GetPolicy(ctx, body.Id)
+	// if err is nil, it means we got a policy, which means we can't create
+	// a new policy with this ID. Return a 409 Conflict http status code
+	// which means the request conflicts with the current state of the server
+	if gpErr == nil {
+		gpErr = auth.ErrAlreadyExists
+		if c.handleAPIError(ctx, w, gpErr) {
+			return
```

Comment:
```suggestion
		if c.handleAPIError(ctx, w, auth.ErrAlreadyExists) {
			return
```
will probably be somewhat easier for me to analyze in future, as I don't need to scan below to see that `gpErr` is not re-used.

Reply:
This code has been moved and refactored per your previous comment.